src/containerapp-preview/azext_containerapp_preview/_clients.py

```diff
@@ -0,0 +1,193 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import json
+
+from azure.cli.core.azclierror import ResourceNotFoundError
+from azure.cli.core.util import send_raw_request
+from azure.cli.core.commands.client_factory import get_subscription_id
+from knack.log import get_logger
+
+from ._constants import GA_CONTAINERAPP_EXTENSION_NAME
+from ._utils import (_get_azext_module)
+
+logger = get_logger(__name__)
+
+PREVIEW_API_VERSION = "2023-04-01-preview"
+HEADER_AZURE_ASYNC_OPERATION = "azure-asyncoperation"
+HEADER_LOCATION = "location"
+
+
+def poll_status(cmd, request_url):
+    azext_client = _get_azext_module(
+        GA_CONTAINERAPP_EXTENSION_NAME, "azext_containerapp._clients")
+    return azext_client.poll_status(cmd, request_url)
+
+
+def poll_results(cmd, request_url):
+    azext_client = _get_azext_module(
+        GA_CONTAINERAPP_EXTENSION_NAME, "azext_containerapp._clients")
+    return azext_client.poll_results(cmd, request_url)
+
+
+class ContainerAppClient(_get_azext_module(GA_CONTAINERAPP_EXTENSION_NAME, "azext_containerapp._clients").ContainerAppClient):
+    @classmethod
+    def get_api_version(cls):
+        return PREVIEW_API_VERSION
+
+
+class ManagedEnvironmentClient(_get_azext_module(GA_CONTAINERAPP_EXTENSION_NAME, "azext_containerapp._clients").ManagedEnvironmentClient):
+    @classmethod
+    def get_api_version(cls):
+        return PREVIEW_API_VERSION
+
+
+class ConnectedEnvironmentClient():
+    @classmethod
+    def get_api_version(cls):
+        return PREVIEW_API_VERSION
+
+    @classmethod
+    def create(cls, cmd, resource_group_name, name, connected_environment_envelope, no_wait=False):
+        management_hostname = cmd.cli_ctx.cloud.endpoints.resource_manager
+        sub_id = get_subscription_id(cmd.cli_ctx)
+        url_fmt = "{}/subscriptions/{}/resourceGroups/{}/providers/Microsoft.App/connectedEnvironments/{}?api-version={}"
+        request_url = url_fmt.format(
+            management_hostname.strip('/'),
+            sub_id,
+            resource_group_name,
+            name,
+            cls.get_api_version())
```

Comment:
If you have new features which are needed to be developed in the future *(especially for atomic commands)*, we strongly recommend that you try our advanced code generation tools [CodeGen - AAZ](https://azure.github.io/aaz-dev-tools/)

This tool is easy to use and can bring you the following benefits:
- It can improve your development efficiency and reduce manual effort.
- It can ensure the style and consistency of code, so it also reduces the burden of PR review.
- It can improve code quality and reduce the occurrence of bugs.
......

If you are interested in it and would like to learn more, please contact our CodeGen expert @kairu-ms

Reply:
Thanks, will consider trying it in the future. The current design follows the already existing module, code already existed.